tests/ecosystem/upgrade/test_logging_upgrade.py

```diff
@@ -0,0 +1,232 @@
+import logging
+
+import pytest
+
+from ocs_ci.ocs.cluster import CephCluster, CephHealthMonitor
+from ocs_ci.ocs.resources.csv import CSV
+from ocs_ci.utility.deployment_openshift_logging import (
+    check_health_of_clusterlogging, get_clusterlogging_subscription
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.packagemanifest import PackageManifest
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocp_upgrade
+)
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import (
+    get_ocp_version, run_cmd
+)
+
+logger = logging.getLogger(__name__)
+
+
+def check_cluster_logging():
+    """
+    Few checks to assert for before and after upgrade
+    1. Checks for pods in openshift-logging project
+    2. Checks for the health of logging cluster before upgrade
+
+    """
+
+    assert get_all_pods(
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    ), 'Some or all pods missing in namespace'
+
+    assert check_health_of_clusterlogging(), "Cluster is not Healthy"
+
+
+@retry(CommandFailed, 5, 30, 2)
+def get_csv_version(channel):
+    """
+    Get the cluster-logging and Elasticsearch CSV version
+    and Images of the pods
+
+    Args:
+        channel (int) : Logging Channel
+
+    Returns:
+        cluster_logging_csv (str) : Name of the cluster-logging CSV
+        elasticsearch_csv (str) : Name of the elasticsearch CSV
+        dict: Images dict like: {'image_name': 'image.url.to:tag', ...}
+
+    """
+
+    clo_package_manifest = PackageManifest(
+        resource_name='cluster-logging'
+    )
+    cluster_logging_csv = clo_package_manifest.get_current_csv(channel)
+    es_package_manifest = PackageManifest(
+        resource_name='elasticsearch-operator'
+    )
+    elasticsearch_csv = es_package_manifest.get_current_csv(channel)
+    logging_csv = CSV(
+        resource_name=cluster_logging_csv,
+        namespace=constants.OPENSHIFT_LOGGING_NAMESPACE
+    )
+    images = ocp.get_images(logging_csv.get())
+    return cluster_logging_csv, elasticsearch_csv, images
+
+
+def check_csv_version_post_upgrade(channel):
+    """
+    This function is to check the CSV version post upgrade
+
+    Args:
+        channel (int) : logging channel
+
+    Returns:
+        bool : Returns True if version matches, False otherwise
+
+    """
+
+    cluster_logging_csv, elasticsearch_csv, images = get_csv_version(channel)
+    if config.UPGRADE['upgrade_logging_channel'] in (
+        cluster_logging_csv and elasticsearch_csv
```

Comment:
This check is probably wrong:
```
("somestring" and "someotherstring")
```
returns `"someotherstring"`.

That means
```
(
    cluster_logging_csv and elasticsearch_csv
)
```
returns `elasticsearch_csv` so `cluster_logging_csv` is omitted entirely.

Shouldn't this be something like this?
```
    if (
        config.UPGRADE['upgrade_logging_channel'] == cluster_logging_csv and
        config.UPGRADE['upgrade_logging_channel'] == elasticsearch_csv
    ):
```

Reply:
done